src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -632,12 +633,21 @@ ReactDOMComponent.Mixin = {
           this._currentElement.type
         );
       }
+      var isCustomComponentTag = isCustomComponent(this._tag, props);
+      if (__DEV__ && isCustomComponentTag && !didWarnShadyDOM && el.shadyRoot) {
+        warning(
+          false,
+          'A component is using shady DOM. Using shady DOM with React can ' +
```

Comment:
I tried using `var name = type.displayName || type.name || 'A component';`, but it did not work since `this._currentElement.type` ended up referring to a string (which I believe is the HTML element tag?) without `displayName` or `name` properties ever